packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2743,28 +2745,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                             });
                     }
                 }
-            } else {
-                if (!this.pendingBatch.push(message)) {
-                    throw new GenericError(
-                        "BatchTooLarge",
+            } else if (!this.pendingBatch.push(message)) {
+                throw new GenericError(
+                    "BatchTooLarge",
                         /* error */ undefined,
-                        {
-                            opSize: (message.contents?.length) ?? 0,
-                            count: this.pendingBatch.length,
-                            limit: this.pendingBatch.options.hardLimit,
-                        });
-                }
-                if (!this.currentlyBatching()) {
-                    this.flush();
-                } else if (!this.flushMicroTaskExists) {
-                    this.flushMicroTaskExists = true;
-                    // Queue a microtask to detect the end of the turn and force a flush.
-                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                    Promise.resolve().then(() => {
-                        this.flushMicroTaskExists = false;
-                        this.flush();
+                    {
+                        opSize: (message.contents?.length) ?? 0,
+                        count: this.pendingBatch.length,
+                        limit: this.pendingBatch.options.hardLimit,
                     });
-                }
+            }
+
+            if (!this.currentlyBatching()) {
```

Comment:
the test at `packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts` surfaced an issue with batches containing only a single attach op with this feature enabled, as the runtime would not have scheduled a flush at the end of the JS turn. The container would be considered 'dirty' and the test would hang because the op did not go over the wire at all.

Reply:
I think we hit it in the past and I even made some fix for it. Or maybe concluded that because attach op alone is not "visible" in any way through public APIs (no way to enumerate data stores), then it's Ok as is.
I think it was the former, and maybe (again) something was wiped out due to merges between main & next?

Here it is: https://github.com/microsoft/FluidFramework/pull/12103

Reply:
> something was wiped out due to merges between main & next?

If it wasn't wiped out explicitly, I'm starting to become concerned about this process.